stream.js

```diff
@@ -67,6 +67,13 @@ function WebSocketStream(target, protocols, options) {
   var coerceToBuffer = options.binary || options.binary === undefined
 
   function socketWriteNode(chunk, enc, next) {
+    // avoid errors, this never happens unless
+    // destroy() is called
+    if (socket.readyState !== WS.OPEN) {
```

Comment:
As I thought this has nothing to do `ws@2` it is also needed for `ws@1`. The same issue also happens on master if you disable the permessage-deflate extension.

```diff
diff --git a/test.js b/test.js
index 18caa28..a0b10af 100644
--- a/test.js
+++ b/test.js
@@ -99,7 +99,9 @@ test('drain', function(t) {
   t.plan(1)
 
   echo.start(function() {
-    var client = websocket(echo.url, echo.options)
+    var client = websocket(echo.url, {
+      perMessageDeflate: false
+    })
 
     client.on('drain', function() {
       client.destroy()
```

In `ws@2` this also happens with permessage-deflate enabled because data is not compressed unless the size is bigger than a configurable threshold (1024 bytes by default).

The problem with the "drain" test is that a single data chunk arrives on the server with many binary frames and a close frame. All frames are parsed on the same tick and when the close frame is processed, the `readyState` is set to `CLOSING`. The payload of the processed binary frames is pushed to the [proxy](https://github.com/maxogden/websocket-stream/blob/92871ebf2b0a3407376c5faec1ffcfd38acc14ab/stream.js#L119), but when [`socketWriteNode()`](https://github.com/maxogden/websocket-stream/blob/92871ebf2b0a3407376c5faec1ffcfd38acc14ab/stream.js#L69-L74) is called the `readyState` is already `CLOSING`.